packages/wonder-stuff-i18n/src/plugins/i18n-plugin.js

```diff
@@ -0,0 +1,410 @@
+            if (
+                (shouldBeLocalized && !localizeRuntimeFiles) ||
+                (!shouldBeLocalized && localizeRuntimeFiles)
+            ) {
+                continue;
+            }
+
+            // Extract the old hash out of the existing JS filename, we'll be
+            // potentially replacing this later.
+            const oldHash = extractHashFromFileName(assetPath);
+            if (!oldHash) {
+                throw new Error(`${assetPath} is missing hash in filename.`);
+            }
+
+            // We pre-parse the strings so that we can re-use the strings
+            // for each locale, this helps to improve performance.
+            const strings = I18nUtils.getI18nStringsFromString(oldSource);
+
+            // Go through each locale that we want to translate into.
+            for (const locale of locales) {
+                let translatedSource = I18nUtils.translateString(
+                    oldSource,
+                    translatedStrings[locale],
+                    strings,
+                );
+
+                // The output location is relative to /prod/en/, so we need
+                // to build a path that points to the correct location.
+                let translatedAssetPath = path.join("..", locale, assetPath);
+
+                // We optionally localize any files which refer to other
+                // JS files, making sure they point to the right location
+                if (shouldBeLocalized) {
+                    translatedSource = localizeFile({
+                        content: translatedSource,
+                        oldHash,
+                        hashMap: hashMap[locale],
+                        locale,
+                        getLocalePath,
+                    });
+                }
+
+                // If the contents of the file have been translated, then we
+                // need to recompute the hash based on the new contents. This
+                // also involves updating the hash in the filename.
+                if (translatedSource !== oldSource) {
+                    const newHash = hashFileContents(translatedSource);
+                    hashMap[locale][oldHash] = newHash;
+                    translatedAssetPath = translatedAssetPath.replace(
+                        oldHash,
+                        newHash,
+                    );
+                }
+
+                // Tell Webpack to write out the translated file
+                addAsset(assets, translatedAssetPath, translatedSource);
+            }
+        }
+
+        if (timing) {
+            timing.end(timingName);
+        }
+
+        return hashMap;
+    }
+
+    /**
+     * Update manifest files so that they point to the correct location of
+     * the translated files.
+     *
+     * @param {assets: Object<string, Object>, hashMap: <string, string>} options
+     */
+    localizeManifests({assets, hashMap}: {assets: Assets, hashMap: HashMaps}) {
```

Comment:
TypeScript supports inlay hints which removes the need for this style of args (except when there's a bunch of optional args in which case an "options" param makes sense).  Having these be separate params would cut down on some of the visualize noise of having to duplicate the field names.  See https://code.visualstudio.com/docs/languages/typescript#_inlay-hints for details about inlay hints (VSCode has a bunch of options to configure in which situations they show up).  There's also a keyboard shortcut to toggle them which is nice.